plugins/kafka/kafka.cpp

```diff
@@ -0,0 +1,1167 @@
+                                    //      STRING      message;
+                                    //  END;
+
+                                    *(__int32*)(row) = messageObjPtr->partition();
+                                    *(__int64*)(row + sizeof(__int32)) = messageObjPtr->offset();
+                                    *(size32_t*)(row + sizeof(__int32) + sizeof(__int64)) = messageObjPtr->len();
+                                    memcpy(row + sizeof(__int32) + sizeof(__int64) + sizeof(size32_t), messageObjPtr->payload(), messageObjPtr->len());
+
+                                    result = rowBuilder.finalizeRowClear(len);
+
+                                    lastMsgOffset = messageObjPtr->offset();
+                                    ++consumedRecCount;
+
+                                    // Give opportunity for consumer to pull in any additional messages
+                                    consumerPtr->handle()->poll(0);
+                                }
+                                break;
+
+                            case RdKafka::ERR__TIMED_OUT:
+                                // No new messages arrived and we timed out waiting
+                                ++attemptNum;
+                                consumerPtr->handle()->poll(timeoutWait);
+                                break;
+
+                            case RdKafka::ERR__PARTITION_EOF:
+                                // We reached the end of the messages in the partition
+                                #ifdef _DEBUG
+                                    DBGLOG("Kafka: EOF reading message from partition %d", messageObjPtr->partition());
+                                #endif
+                                shouldRead = false;
+                                break;
+
+                            case RdKafka::ERR__UNKNOWN_PARTITION:
+                                // Unknown partition; don't throw an error here because
+                                // in some configurations (e.g. more Thor slaves than
+                                // partitions) not all consumers will have a partition
+                                // to read
+                                #ifdef _DEBUG
+                                    DBGLOG("Kafka: Unknown partition while trying to read");
+                                #endif
+                                shouldRead = false;
+                                break;
+
+                            case RdKafka::ERR__UNKNOWN_TOPIC:
+                                throw MakeStringException(-1, "Kafka: Error while reading message: '%s'", messageObjPtr->errstr().c_str());
+                                break;
+                        }
+                    }
+                    catch (...)
+                    {
+                        delete(messageObjPtr);
+                        throw;
+                    }
+
+                    delete(messageObjPtr);
```

Comment:
Good catch.  I've nullified the pointer after the delete and use a new boolean to track message consumption, which is then used within the loop test instead of the pointer check.